### ISSUE TITLE
Amp form always gets "amp-form-submit-success" css class in Gmail

### ISSUE THREAD

**Issue by 06romix** [ampproject/amphtml#24775]
## amp-form doesn't work correctly in Gmail

After submitting form in Gmail and get code 400 from end point - form get css class "amp-form-submit-success" instead "amp-form-submit-error"

## How do we reproduce the issue?

```html
<!doctype html>
<html ⚡4email>
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
  <style amp4email-boilerplate>body{visibility:hidden}</style>
  <style amp-custom>
        .amp-form-submit-success .some-block {
            color: green;
        }
        .amp-form-submit-success .initial-content {
            display: none;
        }
        .amp-form-submit-error {
            color: red;
        }
  </style>
</head> 
<body>
<form method="post"
  action-xhr="https://ampbyexample.com/components/amp-form/submit-form-input-text-xhr">
  <p class="some-block">Form Submission with Page Update</p>
  <div class="initial-content">
    <input type="text"
      name="name"
      placeholder="Name..."
      required>
      <input type="submit"
    value="Subscribe">
  </div>
  <div submit-success>
    <template type="amp-mustache">
      Success! Thanks {{name}} for trying the
      <code>amp-form</code> demo! Try to insert the word "error" as a name input in the form to see how
      <code>amp-form</code> handles errors.
    </template>
  </div>
  <div submit-error>
    <template type="amp-mustache">
      Don't panic! Just enter "cat" as name
    </template>
  </div>
</form>
</body>
</html>
```

1. Send code above from https://amp.gmail.dev/playground/ 
2. Go to gmail
3. Fill form with "error" string
4. Send form

## What browsers are affected?
On Ubuntu

Google Chrome Version 77.0.3865.90 (Official Build) (64-bit)
Firefox 69.0.1

## Which AMP version is affected?

I think it was always broken
Powered by AMP ⚡ HTML – Version 1909141409590

**Comment by caroqliu** [comment]
Hello! Thank you for filing this issue! I am unable to reproduce when following the provided steps. Can you confirm you are still experiencing the issue, and if so, provide additional details such as screenshots?

**Comment by caroqliu** [comment]
Nevermind, I can reproduce. Looking into this.
